crates/test-framework/src/metrics/mod.rs

```diff
@@ -0,0 +1,483 @@
+            .collect::<Vec<_>>();
+        let percentile_99_duration = self
+            .metrics
+            .iter()
+            .map(|metric| metric.percentile_99_duration)
+            .collect::<Vec<_>>();
+        let percentile_90_duration = self
+            .metrics
+            .iter()
+            .map(|metric| metric.percentile_90_duration)
+            .collect::<Vec<_>>();
+        let run_count = self
+            .metrics
+            .iter()
+            .map(|metric| metric.run_count as u64)
+            .collect::<Vec<_>>();
+
+        let extended_metrics_fields = T::fields();
+        let extended_metrics_field_names = extended_metrics_fields
+            .iter()
+            .map(arrow::datatypes::Field::name)
+            .collect::<Vec<_>>();
+
+        let mut columns: Vec<ArrayRef> = vec![
+            Arc::new(StringArray::from(run_id)),
+            Arc::new(StringArray::from(run_name)),
+            Arc::new(StringArray::from(commit_sha)),
+            Arc::new(StringArray::from(branch_name)),
+            Arc::new(StringArray::from(test_type)),
+            Arc::new(UInt64Array::from(started_at)),
+            Arc::new(UInt64Array::from(finished_at)),
+            Arc::new(StringArray::from(query_name)),
+            Arc::new(StringArray::from(query_status)),
+            Arc::new(Float64Array::from(average_duration)),
+            Arc::new(Float64Array::from(median_duration)),
+            Arc::new(Float64Array::from(percentile_99_duration)),
+            Arc::new(Float64Array::from(percentile_90_duration)),
+            Arc::new(UInt64Array::from(run_count)),
+        ];
+
+        if !extended_metrics_fields.is_empty() {
+            let mut extended_metrics_builders = self.build_extended_metrics()?;
+
+            for field in extended_metrics_field_names {
+                match extended_metrics_builders.get_mut(field) {
+                    Some(Builder::String(builder)) => columns.push(Arc::new(builder.finish())),
+                    Some(Builder::UInt64(builder)) => columns.push(Arc::new(builder.finish())),
+                    Some(Builder::Float64(builder)) => columns.push(Arc::new(builder.finish())),
+                    None => {
+                        return Err(anyhow::anyhow!(
+                            "No builder found for extended metric field: {field}"
+                        ))
+                    }
+                }
+            }
+        }
+
+        Ok(vec![RecordBatch::try_new(Self::schema(), columns)?])
+    }
+
+    pub async fn show(&self) -> Result<()> {
+        let schema = Self::schema();
+
+        let records = self.build()?;
+        let ctx = SessionContext::new();
+        let provider = MemTable::try_new(schema, vec![records])?;
+        let df = DataFrame::new(
+            ctx.state(),
+            LogicalPlanBuilder::scan(UNNAMED_TABLE, provider_as_source(Arc::new(provider)), None)?
+                .build()?,
+        );
+
+        df.show().await?;
```

Comment:
There is a utility function in Arrow that we can use to print the batches without needing to pull in DataFusion: https://docs.rs/arrow/latest/arrow/util/pretty/fn.print_batches.html